internetarchive/openlibrary
# Switch to using NJS for nginx IP anonymization

Closes #10139 

See corresponding olsystem PR: https://github.com/internetarchive/olsystem/pull/245

### Technical


### Testing


Tested on ol-www0 ONLY ; not tested on ol-covers0, but should hopefully be alright.

Tested by:

Nginx NJS testing:

Plan:

1. Push to this branch + rebuild olbase with the `fix-nginx-logs` branch https://github.com/internetarchive/openlibrary/actions/workflows/olbase.yaml

1. Pull down the latest fix-nginx-logs image
```sh
docker pull openlibrary/olbase:fix-nginx-logs
```

2. Pull down olsystem changes https://github.com/internetarchive/olsystem/pull/245.diff
```sh
cd /opt/olsystem
git apply <<'EOF'
BLAH
EOF
```

3. And pull down openlibrary changes that need to be on server:
```sh
cd /opt/openlibrary
git apply --reject <<'EOF'
BLAH
EOF
```


2. Restart the nginx container on ol-www0 with the new image
```sh
cd /opt/openlibrary
export COMPOSE_FILE="compose.yaml:compose.production.yaml"
export HOSTNAME="$HOSTNAME"
OLIMAGE="openlibrary/olbase:fix-nginx-logs" docker compose \
    --profile ol-www0 \
    up -d --no-deps \
    web_nginx
```

3. Check if it's starting up

```
docker compose logs --tail=10 -f web_nginx
```

3. Monitor the logs; IPs should be anonymized now
```sh
obfi | head
```


4. Roll back until next deploy
```sh
cd /opt/openlibrary
sudo git stash
export COMPOSE_FILE="compose.yaml:compose.production.yaml"
export HOSTNAME="$HOSTNAME"
docker compose \
    --profile ol-www0 \
    up -d --no-deps \
    web_nginx
```


### Screenshot


No longer dud IPs!

![image](https://github.com/user-attachments/assets/1241ee73-c2e4-4fce-bfb6-b23e9f442d6d)



### Stakeholders
